tests/integ/scala/test_dataframe_suite.py

```diff
@@ -1672,13 +1672,17 @@ def test_createDataFrame_with_given_schema_array_map_variant(session):
     Utils.check_answer(df, expected, sort=False)
 
 
-def test_variant_in_array_and_map(session):
+@pytest.mark.localtest
+def test_variant_in_array_and_map(session, local_testing_mode):
     schema = StructType(
         [StructField("array", ArrayType(None)), StructField("map", MapType(None, None))]
     )
     data = [Row([1, "\"'"], {"a": "\"'"})]
     df = session.create_dataframe(data, schema)
-    Utils.check_answer(df, [Row('[\n  1,\n  "\\"\'"\n]', '{\n  "a": "\\"\'"\n}')])
+    if not local_testing_mode:
+        Utils.check_answer(df, [Row('[\n  1,\n  "\\"\'"\n]', '{\n  "a": "\\"\'"\n}')])
+    else:
+        Utils.check_answer(df, data)
```

Comment:
as we discussed, we shall update the output to json stringfy the object to align with the snowflake behavior

Reply:
updated in connection.py to stringfy the json obj